### PR TITLE
refactor(schema): keep type converters close

### PR DESCRIPTION
**Pull request by zmstone**
Fixes a part of [EMQX-11340](https://emqx.atlassian.net/browse/EMQX-11340)

### <samp>🤖 Generated by Copilot at 13ce7ee</samp>

This pull request introduces a new module `emqx_conf_schema_types.erl` that handles the conversion of typerefl names to different type specs for swagger, dashboard, and docgen. It also refactors the `emqx_conf.erl` and `emqx_dashboard_swagger.erl` modules to use this new module instead of their own logic, to simplify the code and ensure consistency.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [x] Changed lines covered in coverage report
- [n/a] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [n/a] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_conf/src/emqx_conf_schema_types.erl [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+-export([readable/2]).
+-export([readable_swagger/2, readable_dashboard/2, readable_docgen/2]).
+
+%% Takes a typerefl name or hocon schema's display name and returns
+%% a map of different flavors of more readable type specs.
+%% - swagger: for swagger spec
+%% - dashboard: to facilitate the dashboard UI rendering
+%% - docgen: for documenation generation
+readable(Module, TypeStr) when is_binary(TypeStr) ->
```
Comment (thalesmg) [inline apps/emqx_conf/src/emqx_conf_schema_types.erl:27]: Nit: moving this closer to `readable/1` might help navigation.

File: apps/emqx_conf/src/emqx_conf_schema_types.erl [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        docgen => #{type => "String", example => <<"32MB">>}
+    };
+readable("wordsize()") ->
+    #{
+        swagger => #{type => string, example => <<"1024KB">>},
+        dashboard => #{type => 'wordSize'},
+        docgen => #{type => "String", example => <<"1024KB">>}
+    };
+readable("map(" ++ Map) ->
+    [$) | _MapArgs] = lists:reverse(Map),
```
Comment (thalesmg) [inline apps/emqx_conf/src/emqx_conf_schema_types.erl:213]:
```suggestion
    _MapArgs = lists:droplast(Map),
```

File: apps/emqx_conf/src/emqx_conf_schema_types.erl [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+    %%   - Delete this function clause once the above is done.
+    #{
+        swagger => #{type => object},
+        dashboard => #{type => string},
+        docgen => #{type => "Map"}
+    };
+readable("json_binary()") ->
+    #{
+        swagger => #{type => string, example => <<"{\"a\": [1,true]}">>},
+        dashboard => #{type => object},
```
Comment (thalesmg) [inline apps/emqx_conf/src/emqx_conf_schema_types.erl:261]: Q: shouldn't this be a string as well?

File: apps/emqx_conf/src/emqx_conf_schema_types.erl [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+            };
+        _ ->
+            nomatch
+    end.
+
+add_integer_prop(Schema, Key, Value) ->
+    case string:to_integer(Value) of
+        {error, no_integer} -> Schema;
+        {Int, []} when Key =:= minimum -> Schema#{Key => Int};
+        {Int, []} -> Schema#{Key => Int}
```
Comment (thalesmg) [inline apps/emqx_conf/src/emqx_conf_schema_types.erl:334]:
```suggestion
        {Int, []} -> Schema#{Key => Int}
```

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx_conf/src/emqx_conf_schema_types.erl (hunk 2 above)
Comment (zmstone) [reply apps/emqx_conf/src/emqx_conf_schema_types.erl:213]: will fix in the next pr.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**